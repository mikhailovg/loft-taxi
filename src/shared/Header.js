import React from 'react';

import {
    Button,
} from '@material-ui/core';

import logo from '../header-logo.svg';

export const Header = ({setPage}) => {
    return (
        <div className="Header">

            <p className="HeaderLogo">
                <img src={logo} width="156" alt="Logo"/>
            </p>

            <Button color="primary" onClick={() => setPage("map")}>Карта</Button>
            <Button color="primary" onClick={() => setPage("profile")}>Профиль</Button>
            <Button color="primary" onClick={() => setPage("login")}>Вход</Button>
            <Button color="primary" onClick={() => setPage("registration")}>Регистрация</Button>
        </div>
    )
}