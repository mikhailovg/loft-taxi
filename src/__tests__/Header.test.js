import React from 'react';
import ReactDOM from 'react-dom';
import {mount, shallow} from 'enzyme';
import '../setupTests';
import renderer from 'react-test-renderer';

import {Header} from "../shared/Header";

describe('Header', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    const outer = shallow(<Header/>);
    const wrapper = shallow(outer.props().children({ /* context */}));

    it('renders Header', () => {
        expect(wrapper.find('.Header').length === 1).toEqual(true);
    });

    it('renders LoginButton', () => {
        expect(wrapper.find('#LoginButton').exists()).toEqual(true);
    });

    it('don`t renders LogoutButton', () => {
        expect(wrapper.find('#LogoutButton').exists()).toEqual(false);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Header/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

})

// FIX TypeError: window.URL.createObjectURL is not a function
jest.mock('mapbox-gl', () => ({
    Map: () => ({})
}))
