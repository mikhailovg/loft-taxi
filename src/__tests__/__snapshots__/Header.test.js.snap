// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="Header"
>
  <p
    className="HeaderLogo"
  >
    <img
      alt="Logo"
      src="header-logo.svg"
      width="156"
    />
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Карта
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Профиль
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    id="LoginButton"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Войти
    </span>
  </button>
</div>
`;
